src/Support/TypeToSchemaExtensions/ResponderTypeToSchema.php

```diff
@@ -0,0 +1,230 @@
+                if ($item->value instanceof ArrayType) {
+                    $item->value->items = $this->flattenMergeValues($item->value->items);
+
+                    return [$item];
+                }
+
+                if ($item->value->isInstanceOf(JsonResource::class)) {
+                    $resource = $this->getResourceType($item->value);
+
+                    if ($resource->isInstanceOf(MissingValue::class)) {
+                        return [];
+                    }
+
+                    if (
+                        $resource instanceof Union
+                        && (new TypeWalker)->first($resource, fn (Type $t) => $t->isInstanceOf(MissingValue::class))
+                    ) {
+                        $item->isOptional = true;
+
+                        return [$item];
+                    }
+                }
+
+                if (
+                    $item->value instanceof Union
+                    && (new TypeWalker)->first($item->value, fn (Type $t) => $t->isInstanceOf(MissingValue::class))
+                ) {
+                    $newType = array_filter($item->value->types, fn (Type $t) => !$t->isInstanceOf(MissingValue::class));
+
+                    if (!count($newType)) {
+                        return [];
+                    }
+
+                    $item->isOptional = true;
+
+                    if (count($newType) === 1) {
+                        $item->value = $newType[0];
+
+                        return $this->flattenMergeValues([$item]);
+                    }
+
+                    $item->value = new Union($newType);
+
+                    return $this->flattenMergeValues([$item]);
+                }
+
+                if (
+                    $item->value instanceof Generic
+                    && $item->value->isInstanceOf(MergeValue::class)
+                ) {
+                    $arrayToMerge = $item->value->templateTypes[1];
+
+                    // Second generic argument of the `MergeValue` class must be an array.
+                    // Otherwise, we ignore it from the resulting array.
+                    if (!$arrayToMerge instanceof ArrayType) {
+                        return [];
+                    }
+
+                    $arrayToMergeItems = $this->flattenMergeValues($arrayToMerge->items);
+
+                    $mergingArrayValuesShouldBeRequired = $item->value->templateTypes[0] instanceof LiteralBooleanType
+                        && $item->value->templateTypes[0]->value === true;
+
+                    if (!$mergingArrayValuesShouldBeRequired || $item->isOptional) {
+                        foreach ($arrayToMergeItems as $mergingItem) {
+                            $mergingItem->isOptional = true;
+                        }
+                    }
+
+                    return $arrayToMergeItems;
+                }
+
+                return [$item];
+            })
+            ->values()
+            ->all();
+    }
+
+    /**
+     * @param  Generic  $type
+     */
```

Comment:
You can remove the docblock as you're type-hinting.

Reply:
Same as above, Generic extends and implements Type so I need both Hintings